test/unit/receivers/SocketModeReceiver.spec.ts

```diff
@@ -714,13 +713,37 @@ describe('SocketModeReceiver', () => {
       receiver.init(app);
 
       const slackRequestArgs = { body: {}, ack: sinon.fake() };
-
       receiver.client.emit('slack_event', slackRequestArgs);
+
       while (!defaultProcessEventErrorHandlerSpy.called) {
         await delay(50);
       }
-
+      sinon.assert.calledOnce(defaultProcessEventErrorHandlerSpy);
       sinon.assert.notCalled(slackRequestArgs.ack);
     });
+
+    it('does not re-acknowledge events that handle acknowledge and then throw unknown errors', async () => {
+      const processStub = sinon.stub<[ReceiverEvent]>().callsFake(async (event: ReceiverEvent) => {
+        await event.ack();
+        throw new Error('internal error');
+      });
```

Comment:
🤩 Wonderful testcase to have! Thanks for making these improvements and tests read better overall!